library/std/src/sys/xous/os.rs

```diff
@@ -0,0 +1,130 @@
+use super::unsupported;
+use crate::error::Error as StdError;
+use crate::ffi::{OsStr, OsString};
+use crate::fmt;
+use crate::io;
+use crate::marker::PhantomData;
+use crate::path::{self, PathBuf};
+
+#[cfg(not(test))]
+mod c_compat {
+    extern "C" {
+        fn main() -> u32;
+    }
+
+    #[no_mangle]
+    pub extern "C" fn abort() {
+        crate::os::xous::ffi::exit(1);
+    }
+
+    #[no_mangle]
+    pub extern "C" fn _start() {
+        crate::os::xous::ffi::exit(unsafe { main() });
+    }
+
+    // This function is needed by the panic runtime. The symbol is named in
+    // pre-link args for the target specification, so keep that in sync.
+    #[no_mangle]
+    // NB. used by both libunwind and libpanic_abort
+    pub extern "C" fn __rust_abort() -> ! {
+        crate::os::xous::ffi::exit(101);
+    }
+}
+
+pub fn errno() -> i32 {
+    0
+}
+
+pub fn error_string(errno: i32) -> String {
+    core::convert::Into::<crate::os::xous::ffi::Error>::into(errno).to_string()
```

Comment:
`Into` should be in the prelude, right?

Reply:
Good point. I copied that line directly from the compiler, and error messages don't seem to pay attention to the prelude. I've adjusted it to remove the scoping.